### PR TITLE
Thealmarty/core type checker wip

### PR DESCRIPTION
**Pull request by thealmarty**

### CODE REVIEW

File: src/Juvix/Core/MainLang.hs [hunk 1]
```diff
@@ -266,8 +270,22 @@ iType ii g (Free x) =
   case lookup x g of
     Just ann -> return $ snd ann
     Nothing  -> throwError (iTypeErrorMsg ii x)
+--Prim-Const typing rule
 iType _ii _g (Nat _) = return VNats
--- Typing rule not in lang ref atm?
+--App rule, function M applies to N
+iType ii g (App m n) = do
```
Comment (cwgoes) [inline src/Juvix/Core/MainLang.hs]: either this function should take sigma as an argument, or return it

Comment (cwgoes) [reply src/Juvix/Core/MainLang.hs]: concluded: it should return sigma

primitive types should have usage w

**Comment by cwgoes** [comment]
@thealmarty Let me know when I should review and/or merge this.

**Comment by thealmarty** [comment]
Reverted  "Add string name to bound variables." commit to separate it out to another branch.

**Pull request merged**